src/main/java/net/sf/mzmine/modules/visualization/spectra/simplespectra/datapointprocessing/datamodel/customguicomponents/HighlightTreeCellRenderer.java

```diff
@@ -0,0 +1,56 @@
+package net.sf.mzmine.modules.visualization.spectra.simplespectra.datapointprocessing.datamodel.customguicomponents;
+
+import java.awt.Color;
+import java.awt.Component;
+import java.util.logging.Logger;
+import javax.swing.JCheckBox;
+import javax.swing.JComponent;
+import javax.swing.JTree;
+import javax.swing.tree.DefaultTreeCellRenderer;
+import net.sf.mzmine.modules.visualization.spectra.simplespectra.datapointprocessing.datamodel.MSLevel;
+
+public class HighlightTreeCellRenderer extends DefaultTreeCellRenderer {
+
+  private static Logger logger = Logger.getLogger(HighlightTreeCellRenderer.class.getName());
+  /**
+   * 
+   */
+  private static final long serialVersionUID = 1L;
+  private JCheckBox cb;
+  public HighlightTreeCellRenderer(JCheckBox cb) {
+    super();
+    this.cb = cb;
+  }
+
+  @Override
+  public Component getTreeCellRendererComponent(JTree tree, Object value, boolean isSelected,
+      boolean expanded, boolean leaf, int row, boolean hasFocus) {
+
+    JComponent c = (JComponent) super.getTreeCellRendererComponent(tree, value, isSelected,
+        expanded, leaf, row, hasFocus);
+
+    if(!(value instanceof DPPMSLevelTreeNode)) {
+//      logger.finest("rendering value: " + value.toString() + "not important.");
+      setOpaque(false);
+      c.setBackground(getBackgroundNonSelectionColor());
+      return c;
+    }
+//    logger.finest("renderign value: " + value.toString() + " is mslevel node");
+    
+    boolean diffMSn = cb.isSelected();
+    
+    c.setOpaque(true);
+    c.setBackground(getNodeColor((DPPMSLevelTreeNode)value, diffMSn));

```

Comment:
How about just setting setEnabled(false) instead of using the red background?

Reply:
Can you do setEnable(false) on a tree item?
This would be the best way.

Reply:
Tree nodes dont have that functionality, but can make a one, of course. You still need a new renderer though.

Reply:
Every JComponent has setEnabled(), so you can just call c.setEnabled(false) here.

Reply:
https://github.com/mzmine/mzmine2/blob/6be7cc01dcbbe5f53ea7bc167ed3f22a9442bece/src/main/java/net/sf/mzmine/modules/visualization/spectra/simplespectra/datapointprocessing/datamodel/customguicomponents/DisableableTreeCellRenderer.java#L50

Thats what I'm doing, but i did not find the fuctionality to store that information in DefaultMutableTreeNode, thats why i made the DisableableTreeNode (it's toungebreaker, i know :D)